src/Widgets/DataFormOptionsWidget/DataFormOptionsWidgetComponent.tsx

```diff
@@ -61,28 +60,64 @@ provideComponent(DataFormOptionsWidget, ({ widget }) => {
       ) : null}
 
       <div>
-        <select
-          className="form-select"
+        <Select
           defaultValue={defaultValue}
           id={id}
+          isRequired={widget.get('required')}
           name={attributeName ?? ''}
-          required={widget.get('required')}
-        >
-          {widget.get('required') ? (
-            <option value="" disabled>
-              Select an option
-            </option>
-          ) : (
-            <option value=""></option>
-          )}
-
-          {options.map(({ value, title }, index) => (
-            <option value={value} key={[id, 'option', value, index].join('-')}>
-              {title}
-            </option>
-          ))}
-        </select>
+        />
       </div>
     </div>
   )
 })
+
+const Select = connect(
+  function Select({
+    defaultValue,
+    id,
+    isRequired,
+    name,
+  }: {
+    defaultValue: string
+    id: string
+    isRequired: boolean
+    name: string
+  }) {
+    const options = useEnumOptions()
+
+    return (
+      <select
+        className="form-select"
+        defaultValue={defaultValue}
+        id={id}
+        name={name}
+        required={isRequired}
+      >
+        {isRequired ? (
+          <option value="" disabled>
+            Select an option
+          </option>
+        ) : (
+          <option value=""></option>
+        )}
+
+        {options.map(({ value, title }, index) => (
+          <option value={value} key={[id, 'option', value, index].join('-')}>
+            {title}
+          </option>
+        ))}
+      </select>
+    )
+  },
+  {
+    loading: ({ isRequired }: { isRequired: boolean }) => (
+      <div
+        aria-busy="true"
+        className="w-100 loading-placeholder"
+        role="progressbar"
+      >
+        <select className="form-select" required={isRequired} />
+      </div>
```

Comment:
How about extending the `Loading` component with two new options:
```ts
{
  children?: React.ReactNode
  className?: string
}
```

With that the above could reuse `Loading`:

```suggestion
      <Loading className="w-100">
        <select className="form-select" required={isRequired} />
      </Loading>
```